Content.Server/Radiation/Systems/GeigerSystem.cs

```diff
@@ -0,0 +1,187 @@
+using Content.Shared.Hands;
+using Content.Shared.Interaction;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Radiation.Components;
+using Content.Shared.Radiation.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameStates;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Server.Radiation.Systems;
+
+public sealed class GeigerSystem : SharedGeigerSystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly AudioSystem _audio = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly RadiationSystem _radiation = default!;
+
+    private static readonly float ApproxEqual = 0.01f;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<GeigerComponent, ActivateInWorldEvent>(OnActivate);
+
+        SubscribeLocalEvent<GeigerComponent, GotEquippedEvent>(OnEquipped);
+        SubscribeLocalEvent<GeigerComponent, GotEquippedHandEvent>(OnEquippedHand);
+        SubscribeLocalEvent<GeigerComponent, GotUnequippedEvent>(OnUnequipped);
+        SubscribeLocalEvent<GeigerComponent, GotUnequippedHandEvent>(OnUnequippedHand);
+
+        SubscribeLocalEvent<OnRadiationSystemUpdated>(OnUpdate);
+        SubscribeLocalEvent<GeigerComponent, ComponentGetState>(OnGetState);
+    }
+
+    private void OnActivate(EntityUid uid, GeigerComponent component, ActivateInWorldEvent args)
+    {
+        if (args.Handled || component.AttachedToSuit)
+            return;
+        args.Handled = true;
+
+        SetEnabled(uid, component, !component.IsEnabled);
+    }
+
+    private void OnEquipped(EntityUid uid, GeigerComponent component, GotEquippedEvent args)
+    {
+        if (component.AttachedToSuit)
+            SetEnabled(uid, component, true);
+        SetUser(uid, component, args.Equipee);
+    }
+
+    private void OnEquippedHand(EntityUid uid, GeigerComponent component, GotEquippedHandEvent args)
+    {
+        if (component.AttachedToSuit)
+            return;
+
+        SetUser(uid, component, args.User);
+    }
+
+    private void OnUnequipped(EntityUid uid, GeigerComponent component, GotUnequippedEvent args)
+    {
+        if (component.AttachedToSuit)
+            SetEnabled(uid, component, false);
+        SetUser(uid, component, null);
+    }
+
+    private void OnUnequippedHand(EntityUid uid, GeigerComponent component, GotUnequippedHandEvent args)
+    {
+        if (component.AttachedToSuit)
+            return;
+
+        SetUser(uid, component, null);
+    }
+
+    private void OnUpdate(OnRadiationSystemUpdated ev)
+    {
+        var query = EntityQuery<GeigerComponent, RadiationReceiverComponent>();
+        foreach (var (geiger, receiver) in query)
+        {
+            var rads = receiver.CurrentRadiation;
+            SetCurrentRadiation(geiger.Owner, geiger, rads);
+        }
+    }
```

Comment:
Shouldn't this only update geiger counters currently in use? ATM it will iterate all of them so if there's 100 it will dirty all 100 of them which is unnecessary.